### PR TITLE
Added buf keyword to verilog_flex.l

### LINKED ISSUE #977: ODIN_II: buf primitive is not supported
Related issue #848

#### Expected Behaviour
Odin should be able to parse and synthesize verilog files containing any/all synthesizable keywords in the verilog 2005 standard.

#### Current Behaviour
Odin aborts due to a parse error caused by the buf primitive, as this key word is not yet supported.

#### Possible Solution
Add support for the buf keyword

#### Steps to Reproduce
Run Odin II with the verilog code provided by @MohamedElgammal
[top.v.txt](https://github.com/verilog-to-routing/vtr-verilog-to-routing/files/3644361/top.v.txt)
Original Repo: https://github.com/MohamedElgammal/Odin-issues/tree/master/buf_premitive

##### uname -a
```                                                                                                                           ✔  2555  19:19:12
Linux JBpc 5.2.9-arch1-1-ARCH #1 SMP PREEMPT Fri Aug 16 11:29:43 UTC 2019 x86_64 GNU/Linux
```

##### gcc -v
```
gcc version 9.1.0 (GCC) 
```

### PR DESCRIPTION
**Pull request by djns99**
#### Description
Add buf keyword to flex file. Updates test case to use buf keyword

#### Related Issue
Closes #977 

#### Checklist:
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ODIN_II/regression_test/benchmark/task/operators/simulation_result.csv [hunk 1]
```diff
@@ -183,11 +183,7 @@ operators/twobits_logical_less_than/k6_frac_N10_frac_chain_mem32K_40nm         ,
...
-operators/unary_bitwise_bufnode/no_arch                                        , n/a                                   , unary_bitwise_bufnode.blif             , 0   , 10.7        , 3.7          , 0.6                , 100.0           , 1            , 2 , 2  , 6            , 1    , -1   , -1        , -1    , -1     , -1                , 8           , 4           , 6             , 8         
-operators/unary_bitwise_not/k6_frac_N10_frac_chain_mem32K_40nm                 , k6_frac_N10_frac_chain_mem32K_40nm.xml, unary_bitwise_not.blif                 , 0   , 24.0        , 55.9         , 0.4                , 100.0           , 1            , 2 , 2  , 7            , 1    , -1   , -1        , -1    , -1     , 4                 , 8           , 4           , 7             , 9         
+operators/unary_bitwise_not/k6_frac_N10_frac_chain_mem32K_40nm                 , k6_frac_N10_frac_chain_mem32K_40nm.xml, unary_bitwise_not.blif                 , 0   , 24.0        , 55.9         , 0.4                , 100.0           , 1            , 2 , 2  , 7            , 1    , -1   , -1        , -1    , -1     , 4                 , 8           , 4           , 7             , 9
```
Comment (djns99) [inline ODIN_II/regression_test/benchmark/task/operators/simulation_result.csv]: @jeanlego How do I regenerate these results?

Comment (djns99) [inline ODIN_II/regression_test/benchmark/task/operators/simulation_result.csv]: @jeanlego How do I regenerate these results?

Comment (jeanlego) [reply ODIN_II/regression_test/benchmark/task/operators/simulation_result.csv]: I use the Makefile `qor` target but you can run it by hand like so
https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/89ffdc1ff8e7a27e6321926826badd78b8a30ec0/ODIN_II/Makefile#L115-L127

**Comment by djns99** [comment]
@jeanlego @sdamghan I think this is ready to go?

File: ODIN_II/regression_test/benchmark/task/operators/synthesis_result.csv [hunk 2]
```diff
@@ -202,12 +202,12 @@ operators/twobits_logical_less_equal_than/no_arch                              ,
...
-operators/unary_bitwise_not/k6_frac_N10_frac_chain_mem32K_40nm                 , k6_frac_N10_frac_chain_mem32K_40nm.xml, unary_bitwise_not.v --------------------------------------------------------------------                 , 0   , 24.7        , 46.3         , 1.4               , 1            , 2 , 2  , 5            , 1    , 0    , 0         , 0     , -1     , 4                 , 7           , 3           , 5             , 7         
+operators/twobits_logical_less_than/no_arch                                    , n/a                                   , twobits_logical_less_than.v --------------------------------------------------------------------         , 0   , 11.6        , 3.9          , 2.2               , 1            , 5 , 2  , 9            , 1    , -1   , -1        , -1    , -1     , -1                , 8           , 3           , 9             , 11
```
Comment (jeanlego) [inline ODIN_II/regression_test/benchmark/task/operators/synthesis_result.csv]: Can you make sure nothing other than but shows up in the diff?

**Comment by jeanlego** [comment]
thanks for this!

File: ODIN_II/SRC/verilog_flex.l [hunk 3]
```diff
@@ -216,6 +216,7 @@ vPUNCT 	[\?\:\|\^\&\<\>\-\*\/\%\(\)\{\}\[\]\~\!\;\#\,\.\@\=\+]
 <INITIAL>"xnor"						{ MP(); return vXNOR;}
 <INITIAL>"xor"						{ MP(); return vXOR;}
 <INITIAL>"specparam"				{ MP(); return vSPECPARAM;}
+<INITIAL>"buf"				        { MP(); return vBUF;}
```
Comment (jeanlego) [inline ODIN_II/SRC/verilog_flex.l]: I just noticed that line 236 still contains "buf" also and gives off a warning. can you make another PR for this. Thanks
